modules/b2b_logic/logic.c

```diff
@@ -4106,6 +4134,360 @@ int b2bl_bridge(str* key, str* new_dst, str *new_proxy, str* new_from_dname,
+		if (s && s->s && s->len) {
+			e1_dname = s;
+		}
+	}
+	LM_DBG("First entity [%.*s]: To %.*s (Proxy %.*s, Displayname %.*s)\n",
+		e1_id->len, e1_id->s, e1_to->len, e1_to->s,
+		(e1_proxy ? e1_proxy->len : 0), 
+		(e1_proxy ? e1_proxy->s : 0),
+		(e1_dname ? e1_dname->len : 0), 
+		(e1_dname ? e1_dname->s : 0)
+		);
+
+	new_br_ent[0] = b2b_client_new(msg, e1_id, e1_to, e1_proxy, ent1_hnames, ent1_hvals, e1_dname);
+	if (!new_br_ent[0]) {
+		LM_ERR("Failed to create entity 1\n");
+		rc = -1;
+		goto end;
+	}
+
+	list2 = parse_csv_record(ent2);
+	if (!list2) {
+		LM_ERR("Failed to parse CSV record for entitity 2: %.*s\n", ent2->len, ent2->s);
+		rc = -1;
+		goto end;
+	}
+
+	s = &list2->s;
+	if (!s->s || !s->len) {
+		LM_ERR("Failed to parse CSV record for entitity 2: %.*s - no entity name (first parameter)\n", ent2->len, ent2->s);
+		rc = -1;
+		goto end;
+	}
+	e2_id = s;
+
+	s = list2->next ? &list2->next->s : NULL;
+	if (!s || !s->s || !s->len) {
+		LM_ERR("Failed to parse CSV record for entitity 1: %.*s - no to_uri (second parameter)\n", ent2->len, ent2->s);
+		rc = -1;
+		goto end;
+	}
+	e2_to = s;
+
+	s = list2->next->next ? &list2->next->next->s : NULL;
+	if (s && s->s && s->len) {
+		e2_proxy = s;
+		s = list2->next->next->next ? &list1->next->next->next->s : NULL;
+		if (s && s->s && s->len) {
+			e2_dname = s;
+		}
+	}
+	LM_DBG("Second entity [%.*s]: To %.*s (Proxy %.*s, Displayname %.*s)\n",
+		e2_id->len, e2_id->s, e2_to->len, e2_to->s,
+		(e2_proxy ? e2_proxy->len : 0), 
+		(e2_proxy ? e2_proxy->s : 0),
+		(e2_dname ? e2_dname->len : 0), 
+		(e2_dname ? e2_dname->s : 0)
+		);
+
+
+	new_br_ent[1] = b2b_client_new(msg, e2_id, e2_to, e2_proxy, ent2_hnames, ent2_hvals, e2_dname);
+	if (!new_br_ent[1]) {
+		LM_ERR("Failed to create entity 2\n");
+		rc = -1;
+		goto end;
+	}
+
+	hash_index = core_hash(e1_to, e2_to, b2bl_hsize);
+
+	tuple = b2bl_insert_new(msg, hash_index, &init_params,
+		NULL, NULL, -1, &b2bl_key, INSERTDB_FLAG, TUPLE_NO_REPL);
+	if(tuple== NULL)
+	{
+		LM_ERR("Failed to insert new scenario instance record\n");
+		rc = -1;
+		goto end;
+	}
+	tuple->lifetime = 60 + get_ticks();
+	LM_ERR("Key: %.*s (%p)\n", b2bl_key->len, b2bl_key->s, tuple);
+
+	if (params && params->s && params->len > 0) {
+		param_list = parse_csv_record(params);
+		if (!param_list) {
+			LM_ERR("Failed to parse CSV record for Params: %.*s\n", params->len, params->s);
+			rc = -1;
+			goto end;
+		}
+		s = &param_list->s;
+		if (!s->s || !s->len) {
+			LM_ERR("Failed to parse CSV record for params: %.*s - no first parameter\n", params->len, params->s);
+			rc = -1;
+			goto end;
+		}
+		if (s->s[0] == 'n') {
+			tuple->bridge_flags = B2BL_BR_FLAG_NOTIFY | B2BL_BR_FLAG_DONT_DELETE_BRIDGE_INITIATOR;
+			s = param_list->next ? &param_list->next->s : NULL;
+			if (s && s->s && s->len) {
+				remote_tuple = s;
+				s = param_list->next->next ? &param_list->next->next->s : NULL;
+				if (s && s->s && s->len) {
+					if (s->s[0] == 1) remote_tuple_party = 1;
```

Comment:
It seems that this param identifies the old entity that receives the NOTIFY, given as an index in the bridged entities array, so the comparison should probably be `if (s->s[0] == '1')`. Can you also clarify a bit in the docs that this is an extra parameter besides the b2b_logic key?

Reply:
Improved the comparison and improved the docs. Thank you for noticing!